### PR TITLE
New reformatted filesender.py with additional options

### PR DESCRIPTION
**Pull request by KenRoy-Octo**

### CODE REVIEW

File: scripts/client/filesender.py [hunk 1]
```diff
@@ -29,427 +29,429 @@
...
+_PARSER = None
+
+
+################################
+# If error, print usage and exit
+################################
+def print_usage_and_exit(err_message):
+    global _PARSER
+    print("\n" + err_message + "\n")
+    _PARSER.print_help()
```
Comment (WebSpider) [inline scripts/client/filesender.py:69]: This is undefined, there is no help text in your PR

**Comment by monkeyiq** [comment]
I will take more of a look once the missing help text is resolved :)

**Comment by KenRoy-Octo** [comment]
> I will take more of a look once the missing help text is resolved :)

_parser. print_help is a function within _parser that prints out arguments added by ArgumentParser.  It already exists.

See:  https://docs.python.org/3/library/argparse.html

File: scripts/client/filesender.py (hunk 1 above)
Comment (KenRoy-Octo) [reply scripts/client/filesender.py:69]: _parser. print_help is a function within _parser that prints out arguments added by ArgumentParser. It already exists.


https://docs.python.org/3/library/argparse.html

**Comment by KenRoy-Octo** [comment]
_parser.print_help() is being called before it is created.  Fixing now...

**Comment by KenRoy-Octo** [comment]
Fixed.

**Comment by monkeyiq** [comment]
It seems that the updates from https://github.com/filesender/filesender/pull/1396 and https://github.com/filesender/filesender/pull/1358 are removed by the PR?

**Comment by peter-** [comment]
@KenRoy-Octo: Could you integrate more recent changes (see above) and update this PR here?

**Pull request closed without merging**